vms/avm/vm.go

```diff
@@ -482,16 +482,15 @@ func (vm *VM) ParseTx(_ context.Context, bytes []byte) (snowstorm.Tx, error) {
 // Invariant: The context lock is not held
 // Invariant: This function is only called after Linearize has been called.
 func (vm *VM) issueTx(tx *txs.Tx) (ids.ID, error) {
-	vm.ctx.Lock.Lock()
-	defer vm.ctx.Lock.Unlock()
-
-	err := vm.network.IssueTx(context.TODO(), tx)
-	if err != nil {
+	txID := tx.ID()
+	if err := vm.network.IssueTx(context.TODO(), tx); err != nil && !errors.Is(err, mempool.ErrDuplicateTx) {
```

Comment:
nit: can we move `err := vm.network.IssueTx(context.TODO(), tx);` to its own line? This if statement is kinda long...